src/addresses.rs

```diff
@@ -89,12 +93,35 @@ pub fn find_address(
         .next();
 
     match (osm_addr_tag, osm_street_tag) {
-        (Some(addr_tag), Some(street_tag)) => Some(build_new_addr(
-            addr_tag,
+        (Some(house_number_tag), Some(street_tag)) => Some(build_new_addr(
+            house_number_tag,
             street_tag,
             poi,
             geofinder.get(&poi.coord),
         )),
+        (None, Some(street_tag)) => {
+            if let Ok(addrs) = rubber.get_address(&poi.coord) {
+                for addr in addrs.into_iter() {
+                    if let Some(address) = addr.address() {
+                        match address {
+                            mimir::Address::Street(_) => continue,
+                            mimir::Address::Addr(ref a) => {
+                                if a.street.name != *street_tag {
+                                    continue;
+                                }
+                            }
+                        }
+                        return Some(address);
+                    }
+                }
+            }
+            Some(build_new_addr(
+                "",
```

Comment:
I am not sure it makes a big difference, but I think `build_new_addr` should be modified to build a `mimir::Address::Street` variant in this case, instead of `mimir::Address::Addr` that is supposed to be used only when a valid housenumber is present.